### DISCUSSION TITLE
请问“.c++”后缀的文件，怎么当成c++来编译？

### DISCUSSION THREAD

**Discussion by wsw0108** [xmake-io/xmake#1492] [Q&A]
请问：
用了自己仿照protobuf.cpp的rule写的capnp.cpp，生成的文件后缀是".c++“。
暂时不考虑生成后做改名的话，怎么让它被识别为cxx？
(如果在buildcmds:vrunv(capnp...)后去重命名，在第一次运行时会报错说要被重命名的文件找不到）

**Comment by waruqi** [comment]
add_files("src/*.c++", {rules = "c++"}) 或者 target:add("files", "src/*.c++", {rules = "c++"})

**Comment by wsw0108** [comment]
[test-capnp.zip](https://github.com/xmake-io/xmake/files/6781895/test-capnp.zip)

试了下没成功，这里有个工程例子，麻烦看一下，能怎么解决。

改名也可以，不过不知道在哪个时机改不会报文件找不到的错误

**Reply by waruqi** [reply]
不要瞎改，上面的代码没说能吧 rules 加到 compile() 里面去。。

```lua
batchcmds:compile(sourcefile_cp, objectfile, {sourcekind = "cxx", configs = {includedirs = sourcefile_dir, languages = "c++11"}})
```

**Reply by wsw0108** [reply]
感谢~~可以了，还需要改capnp编译出来的文件的std为c++14。（MinGW可以编译通过，msvc又不认识c++文件。。。）

另外，我能在capnp.cpp这样的rule里，得到target所依赖的包capnproto的includedirs么？

这部分弄清楚了，我会把这个capnproto和其他我用到的几个包，都提交到xmake-repo里去。

**Reply by waruqi** [reply]
target:pkg("capnproto"):get("includedirs")

**Reply by wsw0108** [reply]
可以了，只不过get("includedirs")没有值，要get("sysincludedirs")。
target:pkg(xxx)在文档里我没找到~

**Reply by waruqi** [reply]
嗯 现在库默认是走sys
